package/kedro_viz/api/graphql.py

```diff
@@ -126,6 +120,92 @@ def get_all_runs() -> List[Run]:
     return runs
 
 
+def format_run_tracking_data(tracking_data: Dict) -> JSONObject:
+    """Convert tracking data in the front-end format.
+    [{
+        datasetName: 'Data Analysis',
+        datasetType: "kedro.extras.datasets.tracking.metrics_dataset.MetricsDataSet"
+        data: {
+            bootstrap: [
+            { runId: 'My Favorite Sprint', value: 0.8 },
+            { runId: 'Another favorite sprint', value: 0.5 },
+            { runId: 'Slick test this one', value: 1 },
+            ],
+            classWeight: [
+            { runId: 'My Favorite Sprint', value: 23 },
+            { runId: 'Another favorite sprint', value: 21 },
+            { runId: 'Slick test this one', value: 21 },
+            ]
+    }]
+
+    Args:
+        tracking_data: JSON blob of tracking data for selected runs
+    Returns:
+        Dictionary with formatted tracking data for selected runs
+    """
+    tracking_keys = set()
+    for key in tracking_data.keys():
+        for nested_keys in tracking_data[key].keys():
+            tracking_keys.add(nested_keys)
+    runs_tracking_data = {
+        key: [
+            {"runId": run_id, "value": tracking_data[run_id][key]}
+            for run_id in tracking_data
+            if key in tracking_data[run_id]
+        ]
+        for key in sorted(tracking_keys)
+    }
+    return json.loads(json.dumps(runs_tracking_data))
```

Comment:
Why do we have to do this? Can't we just return `runs_tracking_data`?

Reply:
it threw me pylint errors for some reason. but i reckon it should work now that u changed JSONOBject to take a dict. I will give it a go

Reply:
it's still giving me this :( 
 error: Incompatible return value type (got "Dict[Any, List[Dict[str, Any]]]", expected "JSONObject")